samples/account-linking/csharp/README.md

```diff
@@ -24,6 +24,99 @@ Please see the [Code Tours](#code-tours) section for in-depth explanation of the
+To add to this difficulty, AzureAD also does not advise email mapping. Emails and User Principal Names (UPNs) can sometimes change.
+ 
+#### Scenario #2
+If the email addresses are different, you must do the following:
+1.	Ask the user to consent/sign with SSO
+2.	Ask the user to sign in with the email you use with their 3P app
+3.	Associate their Teams identity with their app’s identity in a table (or as an attribute on their database)
+Partners have been having some trouble with both of these scenarios. Scenario #2 is especially problematic for several partners. The underlying difficulty is associating a user’s AAD email and associated access token with a different email. For example, the user’s AAD email could be: john@contoso.com, but the email that is registered with the partner is john@gmail.com or even johnmatthews@contoso.com. In both of these examples, the emails don’t match up. The personal email address issue occurs frequently with apps that exist outside of enterprise scope (e.g., Starbucks, Uber, Coda, Observable). This is especially difficult to process as emails/UPNs can change as well for the same individual. 
+
+**As the solution for Scenario #2 will solve Scenario #1 as well, we will focus on solving Scenario #2. 
+
+## Solution
+### Solution Description
+There is one refresh token associated per user, with a refresh every ~12 hours (time is variable/to be determined).
+As covered earlier, the solution will be targeted to solve Scenario #2 . There are four parts to this solution:
+•	UI to prompt user to login to/accept SSO 
+•	UI to prompt user to login to/accept 3P app
+•	Logic to connect user information from the partner’s auth system to SSO
+•	Table that contains: AzureAD identity (GUID/UUID) – not the AAD access token 
+
+This sample does not store user information from your auth system . Replicating user info can cause a myriad of consistency and data inventory issues. 
+
+Example
+For example, let’s say your auth system uses personal email addresses and its users have no associated Teams identity.
+For this, you would have to build out this entire flow:
+1.	Ask the user to sign in to the app
+2.	Ask you to connect their Teams identity with their app’s identity by going through the SSO flow
+3.	Associate their Teams identity with their app’s identity in a table (or as an attribute on their database
+Which would mean building out three components:
+•	UI to prompt user to login to their Teams SSO after logging in with their Uber credentials
+•	Logic to connect user information from their auth system to SSO
+•	Table that contains: AzureAD identity (GUID/UUID) – not the AAD access token 
+
+
+### Flows
+The flows are:
+Tab:
+1.	User consents to Teams SSO
+2.	User presented with partner auth popup
+3.	Azure AD consent
```

Comment:
This sample wouldn't involve separate Azure AD consent / this is technically step 1